### PR TITLE
[FOR REVIEW] Add ZeroEx Settler trades on Arbitrum

### PR DESCRIPTION
**Pull request by RantumBits**
### Contribution type
Please check the type of contribution this pull request is for:

- [x] New spell(s)
- [ ] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For new spell(s)
If you are building new spell(s), please provide the following information:

- **Spell name(s):** *schema.table_name* zeroex_arbitrum.settler_trades
- **Description:** [Detailed description of the new spell(s) and their purpose] decode 0x API trades on new settler contracts on Arbitrum
- **Who are the new spell(s) for?** [Internal team or general community] both
- **How will the new spell(s) be used downstream?** [Description of downstream usage] merged into dex-agg.trades
- **Implementation details:** [Information on how the spell(s) are implemented]  gets taker and maker amounts only, no fills level data for intermediary swaps 
- **Test instructions:** [How to test the new spell(s)] https://dune.com/queries/3921632
- **Related issue(s):** [Link to related issues, if any]

### CODE REVIEW

File: dex/models/_projects/zeroex/arbitrum/zeroex_arbitrum_settler_trades.sql
Comment (Hosuke) [inline dex/models/_projects/zeroex/arbitrum/zeroex_arbitrum_settler_trades.sql:1]: You can reformat your other spells like this one.

**Review by Hosuke — COMMENTED** [review]
LGTM.✅
Thank you @RantumBits 

[Sample zeroex_arbitrum_settler_trades]()

File: dex/models/_projects/zeroex/arbitrum/zeroex_arbitrum_schema.yml [hunk 1]
```diff
@@ -204,4 +204,54 @@ models:
       - *matcha_limit_order_flag
       - &protocol_fee_paid_eth
         name: protocol_fee_paid_eth
-        description: "The protocol fee paid in ETH"
\ No newline at end of file
+        description: "The protocol fee paid in ETH"
+  - name: zeroex_arbitrum_settler_trades
```
Comment (Hosuke) [inline dex/models/_projects/zeroex/arbitrum/zeroex_arbitrum_schema.yml:208]:
```suggestion
        description: "The protocol fee paid in ETH"

  - name: zeroex_arbitrum_settler_trades
```

File: dex/models/_projects/zeroex/arbitrum/zeroex_arbitrum_settler_trades.sql [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+            {{ source('arbitrum', 'traces') }} AS tr
+        JOIN 
+            result_0x_settler_addresses a ON a.settler_address = tr.to AND a.blockchain = 'arbitrum' AND tr.block_time > a.begin_block_time
+        WHERE 
+            (a.settler_address IS NOT NULL OR tr.to = 0xca11bde05977b3631167028862be2a173976ca11)
+            AND varbinary_substring(input,1,4) IN (0x1fff991f, 0xfd3ad6d4)
+            {% if is_incremental() %}
+                AND {{ incremental_predicate('block_time') }}
+            {% endif %}
+            {% if not is_incremental() %}
```
Comment (jeff-dude) [inline dex/models/_projects/zeroex/arbitrum/zeroex_arbitrum_settler_trades.sql:77]:
```suggestion
            {% else %}
```

File: dex/models/_projects/zeroex/arbitrum/zeroex_arbitrum_settler_trades.sql [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+    JOIN 
+        settler_txs st ON st.tx_hash = logs.tx_hash AND logs.block_time = st.block_time AND st.block_number = logs.block_number
+    WHERE 
+        topic0 IN (0x7fcf532c15f0a6db0bd6d0e038bea71d30d808c7d98cb3bf7268a95bf5081b65,
+        0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef,
+        0xe1fffcc4923d04b559f4d29a8bfc6cda04eb5b0d3c460751c2402c5c5cc9109c)
+        {% if is_incremental() %}
+            AND {{ incremental_predicate('logs.block_time') }}
+        {% endif %}
+        {% if not is_incremental() %}
```
Comment (jeff-dude) [inline dex/models/_projects/zeroex/arbitrum/zeroex_arbitrum_settler_trades.sql:115]:
```suggestion
        {% else %}
```

File: dex/models/_projects/zeroex/arbitrum/zeroex_arbitrum_settler_trades.sql [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+    FROM 
+        {{ source('prices', 'usd') }} AS  pu
+    JOIN 
+        tbl_trades ON (pu.contract_address = taker_token  OR pu.contract_address = maker_token) AND date_trunc('minute',block_time) = minute
+    WHERE 
+        pu.blockchain = 'arbitrum'
+        {% if is_incremental() %}
+            AND {{ incremental_predicate('minute') }}
+        {% endif %}
+        {% if not is_incremental() %}
```
Comment (jeff-dude) [inline dex/models/_projects/zeroex/arbitrum/zeroex_arbitrum_settler_trades.sql:178]:
```suggestion
        {% else %}
```

File: dex/models/_projects/zeroex/arbitrum/zeroex_arbitrum_settler_trades.sql [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+    LEFT JOIN 
+        prices pt ON pt.blockchain = 'arbitrum' AND pt.contract_address = taker_token AND pt.minute = DATE_TRUNC('minute', trades.block_time)
+    LEFT JOIN 
+        prices pm ON pm.blockchain = 'arbitrum' AND pm.contract_address = maker_token AND pm.minute = DATE_TRUNC('minute', trades.block_time)
+    WHERE 
+        1=1 
+        {% if is_incremental() %}
+            AND {{ incremental_predicate('tr.block_time') }}
+        {% endif %}
+        {% if not is_incremental() %}
```
Comment (jeff-dude) [inline dex/models/_projects/zeroex/arbitrum/zeroex_arbitrum_settler_trades.sql:246]:
```suggestion
        {% else %}
```

**Review by jeff-dude — CHANGES_REQUESTED** [review]

**Comment by jeff-dude** [comment]
i will hold off on reviewing the other chains, until we get this one finalized

**Comment by RantumBits** [comment]
> i will hold off on reviewing the other chains, until we get this one finalized

thanks for the review. all changes applied here. i'll update other chains to match

**Comment by RantumBits** [comment]
> > i will hold off on reviewing the other chains, until we get this one finalized
> 
> thanks for the review. all changes applied here. i'll update other chains to match

all now updated to match changes requested here @jeff-dude

**Pull request closed without merging**